git-history.md

```diff
@@ -0,0 +1,69 @@
+# Git History Guidelines
+
+All of our projects follow a *fork + feature branch workflow*. That is, every
+project has a main *upstream* repository (e.g.
+[in-toto/in-toto](https://github.com/in-toto/in-toto)), with a *main* branch,
+called `master` or `develop` (e.g.
+[in-toto/in-toto@develop](https://github.com/in-toto/in-toto/tree/develop)).
+Contributors work on their own *fork* of the *upstream* repository (e.g.
+[lukpueh/in-toto](https://github.com/lukpueh/in-toto)). For every feature or
+fix they are working on, they create a *feature* branch, with a descriptive
+name (e.g.
+[lukpueh/in-toto@threshold-comment-fix](https://github.com/lukpueh/in-toto/tree/threshold-comment-fix)).
+A *feature* branch of a *fork* is incorporated into the *main* branch of the
+*upstream* repository by submitting a *pull request*.
+
+The rest of this document presents some guidelines to keep our projects' git
+histories clean.
+
+__*TL;DR*__
+Avoid merge commits in your *feature* branch. You can use `git rebase` to sync
+your *feature* branch with the *main* branch in *upstream* and also to keep
+your git history clean.
+
+
+__Choose the right base for your *feature* branch__
+In most cases you will base your *feature* branch on the *main* branch of your
+*fork*. Make sure that the *main* branch of your fork is in sync with the
+*main* branch of *upstream* so that you don't work on outdated code.
+Updating the *main* branch of your fork should not require *merge* commits,
+unless you have commited directly to your *main* branch, which you shouldn't!
+
+__Keep the base of your *feature* branch in sync with *upstream*__
+While you are working on your *feature* branch the *main* branch might move
+forward, because some other work was merged into the *main* branch. Check
+regularily, especially before you submit a *pull request*, if the
+base of your *feature* branch has moved forward. If it has, use `git rebase`
+to update the base and fix conflicts as they occur.
```

Comment:
```suggestion
to update the base and fix conflicts as they occur.  (`git fetch --all` and `git rebase <upstream>/<main branch>` instead of `git pull`).  [Here's](https://github.com/theupdateframework/tuf/pull/804#issuecomment-440017361) an example in practice.
```

Reply:
Feel free to reject if the suggestion is too verbose.

Reply:
Cool stuff, I didn't know that commit suggestion feature. Do we want to use it in the future? In general I'm hesitant to let GitHub commit for me ... although I do use it for merging PRs. *OT: Is this something for a new guideline? :)*